### ISSUE TITLE
`citar-file-open-function` should ignore `citar-file-open-external`

### ISSUE THREAD

**Issue by Yevgnen** [bdarcus/citar#465]
**Describe the bug**
`citar-file-open-function` is used to open notes and library files like PDFs. When it's set to `citar-file-open-external`, [this line](https://github.com/bdarcus/citar/blob/main/citar.el#L842) returns `nil` and the file buffer is not opened. And the note template is inserted in current buffer incorrectly.

The `open` command for mac gives 

```
 $ open "/Users/User/org/bibliography/papers/teach-2020-old-dog-new-tricks.org"
The file /Users/User/org/bibliography/papers/teach-2020-old-dog-new-tricks.org does not exist.
Perhaps you meant 'http:///Users/User/org/bibliography/papers/teach-2020-old-dog-new-tricks.org'?
```

I personally think `find-file` is almost enough for `org`, `md` notes for Emacs users. 

**To Reproduce**
1. Set `citar-file-open-function` to `citar-file-open-external`.
2. Open a *new* note.

**Emacs version:**
`GNU Emacs 27.2 (build 1, x86_64-apple-darwin21.1.0, Carbon Version 165 AppKit 2113) of 2021-11-18`

**Additional context**
Tested on macOS, not sure if it's reproducable on other platforms.

**Comment by bdarcus** [comment]
So let me clarify:

Is the real issue that `citar-open-notes` is using `citar-file-open-external`, rather than only `find-file`?

I think this may be an unintended bug with some recent changes.

**Comment by bdarcus** [comment]
@roshanshariff - if you can fix this easily in #460, can you please do that?

Otherwise, I'll do it after I merge that.

We just need open notes always to use `find-file`.

**Comment by bdarcus** [comment]
Should be fixed via #460.
